third-party/realdds/src/dds-device.cpp

```diff
@@ -115,12 +114,13 @@ void dds_device::open( const std::vector< dds_video_stream::profile > & profiles
 
     if ( profiles.size() > device::control::MAX_OPEN_STREAMS )
     {
-        throw std::length_error( "Too many streams to open (" + std::to_string( profiles.size() )
-            + "), max is " + std::to_string( device::control::MAX_OPEN_STREAMS ) );
+        DDS_THROW(length_error, "Too many streams to open (" + std::to_string( profiles.size() )
```

Comment:
Not that it's bad, but I don't see the need to complicate things.
Why do we need a `length_error`, i.e. do we have a use-case where we'd need to distinguish this error from others?

If we were to add another error, I would say also define a dds_exception base class, etc...

For now, I say let's keep it simple and just stick to the one `runtime_error`

Reply:
Conceptually `runtime_error` indicates errors that are detectable only at run-time and `length_error` here is kind of an assertion, indicating a design problem of limiting to MAX_OPEN_STREAMS = 8 (chosen to be larger than currently known number of streams opened in a single RS sensor command ).
See also [stdexcept](https://en.cppreference.com/w/cpp/header/stdexcept) for errors description.

I accept that we can use only `runtime_error` in the name of simplicity, the added value of `length_error` here is not significant.